packages/owl-bot/src/update-lock.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import {newCmd} from './cmd';
+import {core} from './core';
+import * as createPr from './create-pr';
+import {GithubRepo, githubRepoFromUri} from './github-repo';
+import {OctokitFactory} from './octokit-util';
+
+export async function maybeCreatePullRequestForLockUpdate(
+  octokitFactory: OctokitFactory,
+  githubRepo?: GithubRepo,
+  localRepoDir?: string,
+  createPullRequestFromLastCommit = createPr.createPullRequestFromLastCommit,
+  logger = console
+): Promise<void> {
+  const cmd = newCmd(logger);
+  const cwd = localRepoDir ?? '.';
+  const status = cmd('git status --porcelain', {cwd}).toString('utf8').trim();
```

Comment:
does this rely on the fact that an empty string will be returned if no changes occur, I think it might be worth adding a comment that points this out 👍

Reply:
Added a comment.